### PR TITLE
Parallel os execution test

### PR DESCRIPTION
**Pull request by satawatnack**
## Implementation details
Parallel oracle scripts execution at end block (waiting for performance test)

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: benchmark/bench_test.go [hunk 1]
```diff
@@ -30,18 +30,18 @@ var ExecuteCases = map[string]struct {
...
+	// 	parameters: []uint64{1, 10, 1000, 10000, 100000},
+	// },
+	// "allocate_mem": {
+	// 	scenario:   103,
+	// 	parameters: []uint64{1, 10, 1000, 10000, 100000},
+	// },
+	// "find_median": {
+	// 	scenario:   104,
+	// 	parameters: []uint64{0, 1, 10, 50, 100},
+	// },
```
Comment (colmazia) [inline benchmark/bench_test.go:44]: why not test these cases as well?

File: benchmark/bench_test.go [hunk 2]
```diff
@@ -164,7 +164,7 @@ func BenchmarkRequestDataDeliver(b *testing.B) {
 func BenchmarkRequestDataEndBlock(b *testing.B) {
 	for name, tc := range ExecuteCases {
 		for _, pm := range tc.parameters {
-			for _, nr := range []int{0, 1, 5, 10, 20} {
+			for _, nr := range []int{0, 1, 2, 3, 4} {
```
Comment (RogerKSI) [inline benchmark/bench_test.go:167]: please revert back to 0,1,5,10,20

File: x/oracle/abci.go [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
+	"sync"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	abci "github.com/tendermint/tendermint/abci/types"
 
 	"github.com/bandprotocol/chain/v2/x/oracle/keeper"
 	"github.com/bandprotocol/chain/v2/x/oracle/types"
 )
 
+const MAX_CONCURRENT_JOBS = 4
```
Comment (RogerKSI) [inline x/oracle/abci.go:13]: please move it to x/oracle/types/constants.go

Comment (RogerKSI) [reply x/oracle/abci.go:13]: think should change the name to be more specific with resolving requests.

**Review by RogerKSI — CHANGES_REQUESTED** [review]

**Review by colmazia — APPROVED** [review]

**Review by RogerKSI — APPROVED** [review]

**Pull request merged**